records/generator.go

```diff
@@ -81,8 +81,14 @@ func (rg *RecordGenerator) ParseState(leader string, c Config) error {
 		return err
 	}
 
+	hostSpec := labels.HostNameSpec1123
+	if c.EnforceDNS952 {
+		hostSpec = labels.HostNameSpec952
+	}
+	nameSpec := hostSpec.Spec()
+
 	// insert state
-	rg.InsertState(sj, c.Domain, c.SOARname, c.Listener, c.Masters)
+	rg.InsertState(sj, c.Domain, c.SOARname, c.Listener, c.Masters, nameSpec)
```

Comment:
The spec or RFC should be constant across all Inserts so it makes sense for it to be a RecordGenerator field and not an argument to this method.